tests/suites/hup/tests/storagemigration.js

```diff
@@ -0,0 +1,490 @@
+/*
+ * Copyright 2021 balena
+ *
+ * @license Apache-2.0
+ */
+
+'use strict';
+
+const { writeFileSync, unlinkSync } = require('fs');
+const { join } = require('path');
+
+const shouldMigrate = async (that, test) => {
+	// check if both aufs and overlayfs are supported, otherwise skip
+	const storageDriver = await that.context
+		.get()
+		.worker.executeCommandInHostOS(
+			`balena info 2>/dev/null | grep -o -e aufs -e overlay2`,
+			that.context.get().link,
+		);
+	if (storageDriver === "overlay2") {
+		test.comment(`SKIP: Already using overlayfs, skipping migration test.`)
+		return false;
+	}
+	const hasDrivers = await that.context
+		.get()
+		.worker.executeCommandInHostOS(
+			`grep -e overlay -e aufs /proc/filesystems >/dev/null; echo $?`,
+			that.context.get().link,
+		);
+	// 0 means both drivers are supported available, 1 means at least one is missing
+	if (hasDrivers === "1") {
+		test.comment(`SKIP: Both aufs and overlayfs have to be supported to run a migration.`)
+		return false;
+	}
+	return true;
+}
+
+const archiveMigrationLogs = async (that, test) => {
+	try {
+		test.comment(`Archiving migration logs`);
+		const migrationLogFile = join(
+			that.suite.options.tmpdir,
+			'storage-migration.log'
+		);
+		const migrationLogs = await that.context
+				.get()
+				.worker.executeCommandInHostOS(
+					`set -x; journalctl -b all -u balena -t balenad | grep 'to overlay2' || true`,
```

Comment:
I think since #2293 you can just call something like this:
```js
await that.context.get().worker.archiveLogs(
    that.id,
    that.context.get().link,
    "set -x; journalctl -b all -u balena -t balenad | grep 'to overlay2' || true");
```
Can you confirm @vipulgupta2048 ?

Reply:
Indeed, we got the docs right here: https://balena-os.github.io/leviathan/classes/leviathan_worker_helpers.export_.html#archivelogs